nano/node/transport/tcp_listener.cpp

```diff
@@ -32,49 +33,66 @@ nano::transport::tcp_listener::tcp_listener (uint16_t port_a, nano::node & node_
 nano::transport::tcp_listener::~tcp_listener ()
 {
 	// Thread should be stopped before destruction
-	debug_assert (!thread.joinable ());
+	debug_assert (!cleanup_thread.joinable ());
+	debug_assert (!future.valid () || future.wait_for (0s) == std::future_status::ready);
 }
 
 void nano::transport::tcp_listener::start ()
 {
-	debug_assert (!thread.joinable ());
 	debug_assert (!cleanup_thread.joinable ());
+	debug_assert (!future.valid ());
 
 	try
 	{
-		acceptor.open (local.protocol ());
+		asio::ip::tcp::endpoint target{ asio::ip::address_v6::any (), port };
+
+		acceptor.open (target.protocol ());
 		acceptor.set_option (asio::ip::tcp::acceptor::reuse_address (true));
-		acceptor.bind (local);
+		acceptor.bind (target);
 		acceptor.listen (asio::socket_base::max_listen_connections);
 
+		{
+			std::lock_guard<nano::mutex> lock{ mutex };
+			local = acceptor.local_endpoint ();
```

Comment:
Does this need to be strand protected?

Reply:
Up to this point (including `local_endpoint` call) all operations are synchronous, so there should be no need to wrap it in a strand. It would be necessary if it was called after main loop is started, that's why it's cached beforehand.